File: go-client-sdk/pkg/models/shared/codatpublicapimodelsdatadatastatus.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsDataDataStatus struct {
	CurrentStatus          *string    `json:"currentStatus,omitempty"`
	DataType               string     `json:"dataType"`
	LastSuccessfulSync     *time.Time `json:"lastSuccessfulSync,omitempty"`
	LatestSuccessfulSyncID *string    `json:"latestSuccessfulSyncId,omitempty"`
	LatestSyncID           *string    `json:"latestSyncId,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsdatadatastatus.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform } from "class-transformer";


export class CodatPublicApiModelsDataDataStatus extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "currentStatus" })
  currentStatus?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "dataType" })
  dataType: string;

  @SpeakeasyMetadata()
  @Expose({ name: "lastSuccessfulSync" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  lastSuccessfulSync?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "latestSuccessfulSyncId" })
  latestSuccessfulSyncId?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "latestSyncId" })
  latestSyncId?: string;
}